### PR TITLE
Fixes #267

### LINKED ISSUE #267: existsAndIsAccessible returns false for ANY thrown exception
Maybe it should rethrow more exceptions that indicate more severe exceptions?

Issue needs some fleshing out.

### PR DESCRIPTION
**Pull request by nairashwin952013**

### CODE REVIEW

File: java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java [hunk 1]
```diff
@@ -1116,6 +1116,30 @@ public boolean isDirectoryEmpty(final String path) throws IOException {
     public boolean existsAndIsAccessible(final String path) {
         try {
             head(path);
+        } catch (MantaClientHttpResponseException e) {
+                if (e.getServerCode().equals(MantaErrorCode.DIRECTORY_DOES_NOT_EXIST_ERROR)) {
```
Comment (kellymclaughlin) [inline java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java:1120]: This is a nit so feel free to take it or leave, but with an `if/else` block this large it'd probably be a bit cleaner as a `switch` statement.

Comment (nairashwin952013) [reply java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java]: Refactored into `switch` statement.

**Comment by nairashwin952013** [comment]
@kellymclaughlin could use another review pass ?

File: java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java [hunk 2]
```diff
@@ -1117,28 +1117,39 @@ public boolean existsAndIsAccessible(final String path) {
...
-                } else if (e.getServerCode().equals(MantaErrorCode.INTERNAL_ERROR)) {
+                    break;
+                case INTERNAL_ERROR:
                     LOG.error("{} Internal Server error for given path: {}", e.getMessage(), path);
-                } else if (e.getServerCode().equals(MantaErrorCode.SNAPLINKS_DISABLED_ERROR)) {
+                    break;
+                case SNAPLINKS_DISABLED_ERROR:
                     LOG.error("{} Snaplinks disabled for given path: {}", e.getMessage(), path);
-                }
+                    break;
```
Comment (kellymclaughlin) [inline java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java:1151]: Maybe add a default case that logs a generic message?

Comment (nairashwin952013) [reply java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java:1149]: Done.

**Pull request merged**